Utilities/DataFlow/src/TimeframeWriterDevice.cxx

```diff
@@ -0,0 +1,78 @@
+/// @file   TimeframeValidatorDevice.cxx
+/// @author Giulio Eulisse, Matthias Richter, Sandro Wenzel
+/// @since  2017-02-07
+/// @brief  Validator device for a full time frame
+
+#include <thread> // this_thread::sleep_for
+#include <chrono>
+
+#include "DataFlow/TimeframeWriterDevice.h"
+#include "Headers/SubframeMetadata.h"
+#include "Headers/DataHeader.h"
+#include <options/FairMQProgOptions.h>
+
+
+using DataHeader = o2::Header::DataHeader;
+
+using PartPosition = int;
+typedef std::pair<DataHeader, PartPosition> IndexElement;
+
+namespace o2 { namespace DataFlow {
+
+TimeframeWriterDevice::TimeframeWriterDevice()
+  : O2Device{}
+  , mInChannelName{}
+  , mFile{}
+{
+}
+
+void TimeframeWriterDevice::InitTask()
+{
+  mInChannelName = GetConfig()->GetValue<std::string>(OptionKeyInputChannelName);
+  mOutFileName = GetConfig()->GetValue<std::string>(OptionKeyOutputFileName);
+  mFile.open(mOutFileName, std::ofstream::out | std::ofstream::binary);
+}
+
+void TimeframeWriterDevice::Run()
+{
+  while (CheckCurrentState(RUNNING)) {
+    FairMQParts timeframeParts;
+    if (Receive(timeframeParts, mInChannelName, 0, 100) <= 0)
+      continue;
+
+    if (timeframeParts.Size() < 2)
+      LOG(ERROR) << "Expecting at least 2 parts\n";
+
+    auto indexHeader = o2::Header::get<DataHeader>(timeframeParts.At(timeframeParts.Size() - 2)->GetData());
+    // FIXME: Provide iterator pair API for the index
+    //        Index should really be something which provides an
+    //        iterator pair API so that we can sort / find / lower_bound
+    //        easily. Right now we simply use it a C-style array.
+    auto index = reinterpret_cast<IndexElement*>(timeframeParts.At(timeframeParts.Size() - 1)->GetData());
+
+    LOG(INFO) << "This time frame has " << timeframeParts.Size() << " parts.\n";
+    auto indexEntries = indexHeader->payloadSize / sizeof(IndexElement);
+    if (strncmp(indexHeader->dataDescription.str, "TIMEFRAMEINDEX", 14) != 0)
+      LOG(ERROR) << "Could not find a valid index header\n";
+    LOG(INFO) << indexHeader->dataDescription.str << "\n";
+    LOG(INFO) << "This time frame has " << indexEntries << "entries in the index.\n";
+    if ((indexEntries * 2 + 2) != (timeframeParts.Size()))
+      LOG(ERROR) << "Mismatched index and received parts. Expected "
+                 << (timeframeParts.Size() - 2  * 2) << " found " << indexEntries ;
+
+    LOG(INFO) << "Everything is fine with received timeframe\n";
```

Comment:
Yes, I am preparing something which generates a dummy timeframe without having to have to set up the whole chain, to use with the tests, but then I can extend the actual devices a bit. Actually I was thinking we might want to have different split logics, e.g.:
- one timeframe
- as many timeframes as they fit in N GBs

I was thinking also that the reader could use inotify or similar mechanism to read files as they get written to a given directory. Anyways, maybe we should spell out the actual user stories we want to support.